hack/airgapped-demo.sh

```diff
@@ -0,0 +1,262 @@
+    SystemdCgroup = true
+
+# Setup a runtime with the magic name ("test-handler") used for Kubernetes
+# runtime class tests ...
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.test-handler]
+  # same settings as runc
+  runtime_type = "io.containerd.runc.v2"
+  base_runtime_spec = "/etc/containerd/cri-base.json"
+  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.test-handler.options]
+    SystemdCgroup = true
+
+[plugins."io.containerd.grpc.v1.cri"]
+  # use fixed sandbox image
+  sandbox_image = "registry.k8s.io/pause:3.7"
+  # allow hugepages controller to be missing
+  # see https://github.com/containerd/cri/pull/1501
+  tolerate_missing_hugepages_controller = true
+  # restrict_oom_score_adj needs to be true when running inside UserNS (rootless)
+  restrict_oom_score_adj = false
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+  [plugins."io.containerd.grpc.v1.cri".registry.configs]
+    [plugins."io.containerd.grpc.v1.cri".registry.configs."${REGISTRY_ADDRESS}:${REGISTRY_PORT}".tls]
+      insecure_skip_verify = true
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
+      endpoint = ["https://${REGISTRY_ADDRESS}:${REGISTRY_PORT}","https://registry-1.docker.io"]
+    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."*"]
+      endpoint = ["https://${REGISTRY_ADDRESS}:${REGISTRY_PORT}"]
+    # Enable registry.k8s.io as the primary mirror for k8s.gcr.io
+    # See: https://github.com/kubernetes/k8s.io/issues/3411
+    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."k8s.gcr.io"]
+      endpoint = ["https://registry.k8s.io", "https://k8s.gcr.io",]
+EOF
+
+cat <<EOF >"${DEMODATA_DIR}/image-credential-provider-config.yaml"
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: CredentialProviderConfig
+providers:
+- name: shim-credential-provider
+  args:
+  - "get-credentials"
+  - "-c"
+  - "/etc/kubernetes/shim-credential-provider-config.yaml"
+  matchImages:
+  - "*:5000"
+  - "*"
+  - "*.*"
+  - "*.*.*"
+  - "*.*.*.*"
+  - "*.*.*.*.*"
+  - "*.*.*.*.*.*"
+  defaultCacheDuration: "0s"
+  apiVersion: credentialprovider.kubelet.k8s.io/v1beta1
+EOF
+
+cat <<EOF >"${DEMODATA_DIR}/shim-credential-provider-config.yaml"
+apiVersion: config.kubeletimagecredentialprovidershim.d2iq.com/v1alpha1
+kind: KubeletImageCredentialProviderShimConfig
+mirror:
+  endpoint: "${REGISTRY_ADDRESS}:${REGISTRY_PORT}"
+  credentialsStrategy: MirrorCredentialsFirst
+credentialProviderPluginBinDir: /etc/kubernetes/image-credential-provider/
+credentialProviders:
+  apiVersion: kubelet.config.k8s.io/v1beta1
+  kind: CredentialProviderConfig
+  providers:
+  - name: static-mirror-credential-provider
+    matchImages:
+    - "${REGISTRY_ADDRESS}:${REGISTRY_PORT}"
+    defaultCacheDuration: "0s"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1beta1
+  - name: static-docker-io-credential-provider
+    matchImages:
+    - "docker.io"
```

Comment:
I wonder how we can do this with all other public registries, maybe something like this at the end of the list. But then the shim plugin would need to know about this and update the response with the actual registry from the image.
```suggestion
  - name: static-empty-credential-provider
     matchImages:
     - "*"
     - "*.*"
     - "*.*.*" 
     - "*.*.*.*"
     - "*.*.*.*.*"
     - "*.*.*.*.*.*"
```

Reply:
I think that should be handled explicitly in the shim config as you suggest. Perhaps this could be the default in the config?